frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/BatchAggregator.java

```diff
@@ -42,13 +43,17 @@ public BaseModelRequest getRequest(String threadName) throws InterruptedExceptio
         // first job is a blocking call;
         Job job = model.nextJob(threadName);
         if (job.isControlCmd()) {
-            RequestInput input = job.getPayload();
-            int gpuId = -1;
-            String gpu = input.getStringParameter("gpu");
-            if (gpu != null) {
-                gpuId = Integer.parseInt(gpu);
+            if (job.getCmd().equals(WorkerCommands.STOP)) {
```

Comment:
You don't need a stop command to break this loop.
currentThread.interrup() will break it. You only need to call shutdown()

Reply:
I couldn't figure how to call interrupt() on runnable. The currentThread context clears a lot of code.